### PR TITLE
Add a test framework for monitoring the host system from inside a container

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?
Closes https://github.com/elastic/beats/issues/38241

This adds a lightweight test framework that runs a set of system tests under a container with the goal of monitoring the host system. The goal with these tests is to catch the numerous edge cases that happen when the system metrics function from a `/hostfs` path inside a container.

The tests have a fairly large matrix of configurations, as we need to test both a wide variety of container permission settings, as well as differences in how linux distros will configure cgroups.

The framework here was designed with the goal of being relatively idiomatic; you can just run the framework with `go test` as you would normally.

You can run the tests yourself with `go test -v ./tests`

As you may have noticed, there's a non-zero amount of TODO statements here, since these tests were built to aggravate a bunch of existing bugs, so certain parts of the tests will remain un-implemented until those bugs are fixed.

## Why is it important?

See https://github.com/elastic/beats/issues/38241, we really need test for this particular case.

## List of bugs that are responsible for TODO statements in the tests:

- https://github.com/elastic/elastic-agent-system-metrics/issues/141
- https://github.com/elastic/elastic-agent-system-metrics/issues/135
- https://github.com/elastic/elastic-agent-system-metrics/issues/139
- https://github.com/elastic/elastic-agent-system-metrics/issues/132
- https://github.com/elastic/go-sysinfo/issues/12

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.md`

### CODE REVIEW

File: dev-tools/systemtests/container.go [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+	// skip if the process has already been created
+	if tr.CreateHostProcess != nil && tr.CreateHostProcess.Process == nil {
+		// We don't need to do this in a channel, but it prevents races between this goroutine
+		// and the rest of test framework
+		startPid := make(chan int)
+		log.Infof("Creating test Process...")
+		go func() {
+			err := tr.CreateHostProcess.Start()
+			// if the process fails to start up, the resulting tests will fail, so just log it
+			require.NoError(tr.Runner, err, "error starting monitor process")
```
Comment (belimawr) [inline dev-tools/systemtests/container.go:292]: `require` uses `t.FailNow` which should not be called from a different goroutine than the one running the test, from the [docs](https://pkg.go.dev/testing#T.FailNow):
> (...) FailNow must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test. Calling FailNow does not stop those other goroutines.

It's better to use `assert` here
```suggestion
			assert.NoError(tr.Runner, err, "error starting monitor process")
```

File: metric/system/process/process.go [hunk 2]
```diff
@@ -280,6 +280,7 @@ func (procStats *Stats) pidFill(pid int, filter bool) (ProcState, bool, error) {
 
 	// network data
 	if procStats.EnableNetwork {
+		// TODO: this can't handle hostfs settings
```
Comment (belimawr) [inline metric/system/process/process.go:283]: Does this TODO still need to be done? Or can this comment be removed?

Comment (fearful-symmetry) [reply metric/system/process/process.go:283]: ...I need to file a bug for it, thanks for reminding me

File: .buildkite/scripts/system-container-tests.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+source .buildkite/scripts/common.sh
+
+install_go_dependencies
+
+# install docker
+
+# DOCKER_VERSION="25.0"
+
+# curl -fsSL https://get.docker.com -o get-docker.sh
+# sudo sh ./get-docker.sh --version $DOCKER_VERSION
```
[the remark this answers is not in the document]
Comment (fearful-symmetry) [reply .buildkite/scripts/system-container-tests.sh:14]: Depending on how creative I get with the buildkite images later, I might need it, so I'm leaving it in

**Review by faec — APPROVED** [review]
Looks good!

**Review by leehinman — APPROVED** [review]
